test/src/tpm/tpm2_interface_test.c

```diff
@@ -749,3 +761,104 @@ void test_rollNonces(void)
   CU_ASSERT(rollNonces(&session, new) != 0);
 
 }
+
+
+
+//----------------------------------------------------------------------------
+// test_unseal_apply_policy
+//----------------------------------------------------------------------------
+void test_unseal_apply_policy(void)
+{
+  TSS2_SYS_CONTEXT *sapi_ctx = NULL;
+
+  init_tpm2_connection(&sapi_ctx);
+  TPML_PCR_SELECTION pcrs_struct = {.count = 0, };
+
+
+  TPM2B_DIGEST policy1;
+  TPM2B_DIGEST policy2;
+
+  policy1.size = 0;
+  policy2.size = 0;
+
+  int pcrs[1] = { };
+  pcrs[0] = 23;
+  init_pcr_selection(sapi_ctx, pcrs, 1, &pcrs_struct);
+  CU_ASSERT(create_policy_digest(sapi_ctx, pcrs_struct, &policy1) == 0);
+  CU_ASSERT(policy1.size != 0);
+  //Remove printf and tpm2_pcrread after review
+  printf("\nPolicy1: %X\n", policy1.buffer);
+  system("tpm2_pcrread sha256:23");
+
+  system("tpm2_pcrextend 23:sha256=0000000000000000000000000000000000000000000000000000000000000001");
+  init_pcr_selection(sapi_ctx, pcrs, 1, &pcrs_struct);
+  CU_ASSERT(create_policy_digest(sapi_ctx, pcrs_struct, &policy2) == 0);
+  CU_ASSERT(policy2.size != 0);
+  //Remove printf and tpm2_pcrread after review
+  printf("Policy2: %X\n", policy2.buffer);
+  system("tpm2_pcrread sha256:23");
+
+  SESSION unsealData_session;
+  CU_ASSERT(create_auth_session(sapi_ctx, &unsealData_session, TPM2_SE_POLICY) == 0);
+  init_pcr_selection(sapi_ctx, pcrs, 1, &pcrs_struct);
+  CU_ASSERT(unseal_apply_policy(sapi_ctx, unsealData_session.sessionHandle, pcrs_struct, policy1, policy2) == 0);
+  system("tpm2_pcrreset 23");
+  //Remove tpm2_pcrread after review
+  system("tpm2_pcrread sha256:23");
+
+  free_tpm2_resources(&sapi_ctx);
+}
+
+//----------------------------------------------------------------------------
+// test_apply_policy_or
+//----------------------------------------------------------------------------
+void test_apply_policy_or(void)
+{
+  TSS2_SYS_CONTEXT *sapi_ctx = NULL;
+
+  init_tpm2_connection(&sapi_ctx);
+  TPML_PCR_SELECTION pcrs_struct = {.count = 0, };
+
+  TSS2L_SYS_AUTH_COMMAND const *nullCmdAuths = NULL;
+  TSS2L_SYS_AUTH_RESPONSE *nullRspAuths = NULL;
+
+  TPM2B_DIGEST policy1;
+  TPM2B_DIGEST policy2;
+  TPM2B_DIGEST policyOR;
+
+  policy1.size = 0;
+  policy2.size = 0;
+  policyOR.size = 0;
+
+  int pcrs[1] = { };
+  pcrs[0] = 23;
+  init_pcr_selection(sapi_ctx, pcrs, 1, &pcrs_struct);
+  CU_ASSERT(create_policy_digest(sapi_ctx, pcrs_struct, &policy1) == 0);
+  CU_ASSERT(policy1.size != 0);
+  //Remove printf and tpm2_pcrread after review
+  printf("\nPolicy1: %X\n", policy1.buffer);
```

Comment:
This in not part of what will be merged and doesn't need to be changed, but the 8-byte value printed here is not the 32-byte digest. There may be a more elegant way to do it, but I think something like this will work:

```  printf("\nPolicy1: 0x");
  for (int i = 0; i < policy1.size; i++)
  {
    printf("%02X", policy1.buffer[i]);
  }
  printf("\n");
```